docs/arch/adr-31.md

```diff
@@ -0,0 +1,90 @@
+# ADR 31: RotatingRectangle Tool and TemporalRotatingRectangle Tool
+
+## Status
+
+In Review
+
+## Context
+
+The Solar Jets project requires a TemporalRotatingRectangle drawing tool for volunteers to capture the width, height, angle and time properties of a solar jet. At the time of this writing, the `front-end-monorepo` has a standard Rectangle drawing tool which gives the width and height properties. Based on the standard Rectangle drawing tool, we want to create a RotatingRectangle drawing tool and a TemporalRotatingRectangle drawing tool.
+
+The RotatingRectangle drawing tool is a common tool used in the Panoptes code based and eventually needs to be written in the `front-end-monorepo`. This is a good opportunity to complete this task.
+
+The idea is to create two new rectangle drawing tools based off the standard Rectangle drawing tool. once complete, the `front-end-monorepo` will have the following rectangle drawing tools:
+
+**Rectangle**
+
+```json
+{
+  "details": array,
+  "frame": number,
+  "height": number,
+  "toolIndex": number,
+  "toolType": string,
+  "width": number,
+  "x_center": number,
+  "Y_center": number
+}
+```
+
+**RotatingRectangle**
+
+\*everything the `Rectangle` tool has plus this addition:
+
+```json
+{
+  "angle": number
+}
+```
+
+**NOTE:** angle is Degrees (-180, 180)
+
+- positive angles rotate from +x to +y (a right-handed coordinate system)
+- rotation is about the center of the rectangle
+
+**TemporalRotatingRectangle**
+
+\*everything the `RotatingRectangle` tool has plus this addition:
+
+```json
+{
+  "displayTime": string
+}
+```
+
+**NOTE:** displayTime will be formatted as:
+
+- If minutes exist: mm:ss:ms
```

Comment:
Is the implication here that there could be video subjects that are minutes long that projects want volunteers to classify on? I hesitate to support this. We found in past projects that volunteers lost interest on 30 second videos plus the longer the video, the larger the file size will be per subject. Chimp & See found the sweet spot to be around 10 seconds. What are the requirements for Solar Jets? If we can limit this to `ss:ms`, then I think we should.

Reply:
I am trying to cover a wider range of video lengths while keeping the video controller UI clean for shorter videos. This is already built in and lives in the `FormattedTime.js` file.

Reply:
Ok. I think in all likelihood based on our file size limitations that we won't have subjects with minutes.